File: cart/src/main/java/com/application/cart/model/CartItem.java
package com.application.cart.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@RedisHash( "CART_ITEM" )
public class CartItem implements Serializable {
    @Serial
    private static final long serialVersionUID = 5811817633994165060L;
    private String id;
    private Long userId;
    private Long productId;
    private Integer amount;
    private BigDecimal productPrice;

}

File: MobileShop/src/app/models/cart-item.ts
import { Product } from './product';

export class CartItem {

    id: string;
    userId: number;
    productId: number;
    productName: string;
    amount: number;
    productPrice: number;
    picture: string;

    constructor(product: Product) {
        this.productId = product.id;
        this.userId = product.userId;
        this.picture = product.picture;
        this.productPrice = product.price;
        this.amount = product.amount;
        this.productName = product.name;
    }
}
